1 Quick Sort/QuickSort1.py

```diff
@@ -0,0 +1,83 @@
+import random
+import time
+import os
+
+def readArrayFromFile(path):
+    numbers = []
+    try:
+        file = open(path,'r')
+        text = ""
+        for line in file:
+            text = text + line
+        numbers = toIntArray(text.split())
+        print(len(numbers))
+        return numbers 
+    except FileNotFoundError:
+        print("File does not exist")
+        return[] 
+def saveSortedArr(array, path):
+        fileName = os.path.basename(path)
+        if array == ascendingSortedArray:
+                fileName = reverseFileName(fileName)
+                file = open((fileName+".py"),"tw")
+                file.write(str(array))
+                file.close()
+        elif array == desendingSortedArray:
+                fileName = "Sorted " + fileName + ".py"
+                file = open((fileName),"tw")
+                file.write(str(array))
+                file.close()
+
+def reverseFileName(string):
+        invertedString = ""
+        length = len(string)
+        for index in range(3, length):
+                invertedString = invertedString + string[length-index-1]
+        return invertedString
+
+def toIntArray(array):
+    intArray = []
+    for symbol in array:
+        digit = int(symbol)
+        intArray.append(digit)
+    return  intArray
+
+def quicksort(array, direction):
+    if len(array) <= 1:
+        return array
+    else:
+        randomNumber = random.choice(array)
+        s = []
+        m = []
+        e = []
+        for element in array:
+            if element < randomNumber:
+                s.append(element)
+            elif element > randomNumber:
+                m.append(element)
+            else:
+                e.append(element)
+        if direction == "asc":
+            return quicksort(s, "asc") + e + quicksort(m, "asc")
+        else:
+            return quicksort(m, "des") + e + quicksort(s, "des")
+
+numbers = []
+path = input("Enter path to the file:")
+numbers = readArrayFromFile(path)
+choice = 0
+while numbers == []:
+    choice = input("Incorrect path.Try again?\nY/N:")
+    if choice == "N":
+        break
+    elif choice == "Y":
+        path = input("Enter path to the file:")
+        numbers = readArrayFromFile(path)
+if choice != "N":                                
```

Comment:
Program breaks is file is empty

Reply:
Please mark all the passed tests and resolve conversation: 

- [ ] Give wrong file name
- [ ] Empty file
- [ ] File with text instead of array numbers
- [ ] Huge number in file like 999999999999999999999999999999999999999999999999999999999